### ISSUE TITLE
apolloClient.cache.reset().then(...

### ISSUE THREAD

**Issue by Enalmada** [apollographql/reason-apollo#158]
I am trying to migrate [this](https://github.com/zeit/next.js/blob/canary/examples/with-apollo-auth/components/SigninBox.js#L26) apollo auth example to reason and am unsure how to do `apolloClient.cache.reset().then()`.  I assume most people using apollo need to clear cache when login status changes?  Could someone advise on if this is currently possible?  

Current workaround:
```
   %bs.raw {| apolloClient.cache.reset().then(() => {require("next/router").default.replace("/")} ) |};
```

**Issue Labels**

- [x] feature

**Comment by Enalmada** [comment]
Actually from reading the apollo docs it seems like `apolloClient.resetStore()` is what should be used.
https://www.apollographql.com/docs/react/recipes/authentication.html#login-logout

It does seem like there is a `resetStore` in the apollo client binding:
https://github.com/apollographql/reason-apollo/blob/d158d97d73d0e35597f0c1e420c3160f822bf44d/src/ApolloClient.re#L19
Could/should it have a Js.promise.t on it to be able to do `.then()`?
```
  "resetStore": [@bs.meth] (unit => Js.Promise.t(unit)),
```

**Comment by Enalmada** [comment]
@Gregoirevda could you advise what kind of promise return type `apolloClient.resetStore` should have?

**Comment by grsabreu** [comment]
Js.Promise.t(unit). Would you mind send a PR for us? Thanks

**Comment by Gregoirevda** [comment]
Fix published under 0.15.1
